client/src/common/hooks/useSocketProvider.js

```diff
@@ -0,0 +1,239 @@
+import { useCallback, useEffect, useMemo } from 'react';
+import { useQuery, useQueryClient } from '@tanstack/react-query';
+
+import {
+  FEAT_CUESHEET,
+  FEAT_EVENTLIST,
+  FEAT_INFO,
+  FEAT_MESSAGECONTROL,
+  FEAT_PLAYBACKCONTROL,
+  TIMER,
+} from '../api/apiConstants';
+import { useSocket } from '../context/socketContext';
+
+export const useEventListProvider = () => {
+  const { data } = useQuery(FEAT_EVENTLIST, () => undefined, {
+    cacheTime: Infinity,
+    staleTime: Infinity,
+  });
+  const placeholder = useMemo(
```

Comment:
if you want to have the same object always, you could just declare the placeholder outside of the hook. No need for `useMemo` here.